app-emulation/genymotion-bin/genymotion-bin-3.2.1-r1.ebuild

```diff
@@ -0,0 +1,122 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit bash-completion-r1 desktop pax-utils
+
+MY_PN="${PN/-bin}"
+MY_P="${MY_PN}-${PV}"
+BIN_ARCHIVE="${MY_P}-linux_x64.bin"
+
+DESCRIPTION="Complete set of tools that provide a virtual environment for Android"
+HOMEPAGE="https://genymotion.com"
+SRC_URI="${BIN_ARCHIVE}"
+
+LICENSE="genymotion"
+SLOT="0"
+KEYWORDS="-* ~amd64"
+
+RDEPEND="app-emulation/virtualbox
+	|| (
+		dev-libs/openssl-compat:1.0.0
+		=dev-libs/openssl-1.0*:0
+	)
+	>=dev-libs/hiredis-1.0.0
+	sys-apps/util-linux
+	virtual/opengl
+"
+BDEPEND="dev-util/patchelf"
+
+RESTRICT="bindist fetch"
+S="${WORKDIR}"
+
+QA_PREBUILT="
+	opt/${MY_PN}/*.so*
+	opt/${MY_PN}/imageformats/*.so
+	opt/${MY_PN}/plugins/*.so*
+	opt/${MY_PN}/${MY_PN}
+	opt/${MY_PN}/genyshell
+	opt/${MY_PN}/player
+	opt/${MY_PN}/${MY_PN}adbtunneld
+	opt/${MY_PN}/gmtool
+"
+
+pkg_nofetch() {
+	einfo
+	einfo "Please visit"
+	einfo
+	einfo "  https://www.genymotion.com/download/"
+	einfo
+	einfo "and download "
+	einfo
+	einfo "  ${BIN_ARCHIVE}"
+	einfo
+	einfo "which must be placed in DISTDIR directory."
+	einfo
+}
+
+src_unpack() {
+	cp "${DISTDIR}/${BIN_ARCHIVE}" "${WORKDIR}" || die "cp failed"
+}
+
+src_prepare() {
+	default
+
+	chmod +x ${BIN_ARCHIVE} || die "chmod failed"
+	yes | ./${BIN_ARCHIVE} > /dev/null || die "unpack failed"
+
+	# removed windows line for bashcompletion
+	sed -i "/complete -F _gmtool gmtool.exe/d" "${MY_PN}/completion/bash/gmtool.bash" || die "sed failed"
+
+	# patch to support hiredis
+	for i in genymotion genyshell gmtool player libcom.so.1.0.0 librendering.so.1.0.0 ; do
```

Comment:
Hiredis didn't mean to bump SONAME here, so that's been fixed now in 1.0.2, but FWIW, this approach is rather dangerous and I'm not keen on doing it in general.

There's no real way to be sure it's safe and I'd rather pin deps on the specific subslot of the thing that broke ABI (given rebuilds aren't possible).

Thank you for the PR though!